analyzer/tests/unit/test_checker_handling.py

```diff
@@ -94,14 +119,131 @@ def test_no_disabled_checks(self):
+
+        # "security" profile, but alpha -> not in default.
+        security_profile_alpha = [
+                'alpha.security.ArrayBound',
+                'alpha.security.MallocOverflow']
+
+        # "default" profile.
+        default_profile = [
+                'security.FloatLoopCounter',
+                'deadcode.DeadStores']
+
+        # Checkers covering some "sei-cert" rules.
+        cert_guideline = [
+                'alpha.core.CastSize',
+                'alpha.core.CastToStruct']
+
+        checkers = []
+        checkers.extend(map(add_description, security_profile_alpha))
+        checkers.extend(map(add_description, default_profile))
+        checkers.extend(map(add_description, cert_guideline))
+
+        # "default" profile checkers are enabled explicitly. Others are in
+        # "default" state.
+        cfg_handler = ClangSA.construct_config_handler(args, context)
+        cfg_handler.initialize_checkers(context, checkers)
+        self.assertTrue(all_with_status(CheckerState.enabled)
+                        (cfg_handler.checks(), default_profile))
+        self.assertTrue(all_with_status(CheckerState.default)
+                        (cfg_handler.checks(), security_profile_alpha))
+
+        # "--enable-all" leaves alpha checkers in "default" state. Others
+        # become enabled.
+        cfg_handler = ClangSA.construct_config_handler(args, context)
+        cfg_handler.initialize_checkers(context, checkers, enable_all=True)
+        self.assertTrue(all_with_status(CheckerState.default)
+                        (cfg_handler.checks(), security_profile_alpha))
+        self.assertTrue(all_with_status(CheckerState.enabled)
+                        (cfg_handler.checks(), default_profile))
+
+        # Enable alpha checkers explicitly.
+        cfg_handler = ClangSA.construct_config_handler(args, context)
+        cfg_handler.initialize_checkers(context, checkers, [('alpha', True)])
+        self.assertTrue(all_with_status(CheckerState.enabled)
+                        (cfg_handler.checks(), security_profile_alpha))
+        self.assertTrue(all_with_status(CheckerState.enabled)
+                        (cfg_handler.checks(), default_profile))
+
+        # Enable "security" profile checkers.
+        cfg_handler = ClangSA.construct_config_handler(args, context)
+        cfg_handler.initialize_checkers(context, checkers,
+                                        [('profile:security', True)])
+        self.assertTrue(all_with_status(CheckerState.enabled)
+                        (cfg_handler.checks(), security_profile_alpha))
+        self.assertTrue(all_with_status(CheckerState.enabled)
+                        (cfg_handler.checks(), default_profile))
+
+        # Enable "security" profile checkers without "profile:" prefix.
+        cfg_handler = ClangSA.construct_config_handler(args, context)
+        cfg_handler.initialize_checkers(context, checkers,
+                                        [('security', True)])
```

Comment:
If I enable the security checker with the `-e security` option I will see the following warning message:
```
[WARNING 2020-07-31 09:55] - Profile name 'security' conflicts with a checker(-group) name.
```
How can I hide this message when I want to enable the security profile and the security checker?

Reply:
You can't. The problem is that `security` can't mean checker group. My personal preference would be forcing users to use `profile:` label in case of profiles and "no label" would always mean a checker(group). I believe this will never happen, because don't like breaking legacy behaviors.